src/utils.js

```diff
@@ -1,22 +1,44 @@
 const CALLBACK_NAME = '_grecaptchaonloadcallback'
 
+let loadScriptPromise
+
+// Returns a promise that resolves once Google's reCAPTCHA library is loaded. If the library is already loaded, then no
+// work is performed, otherwise the library is dynamically loaded.
+// @param {string} [locale] - one of the language codes at https://developers.google.com/recaptcha/docs/language.
+// @returns {Promise}
 export function loadScript(locale) {
-  return new Promise((resolve, reject) => {
-    const url = `https://www.google.com/recaptcha/api.js?onload=${encodeURIComponent(
-      CALLBACK_NAME
-    )}${locale === false ? '' : `&hl=${encodeURIComponent(locale)}`}`
+  // Do this work just once, regardless of how many times this function is called, by saving the created promise and
+  // returning it forevermore.
+  if (!loadScriptPromise) {
```

Comment:
Could you add and early return here to remove the `else` statement?

```js
if (loadScriptPromise) {
  return loadScriptPromise
}
```

Reply:
It wouldn't remove any else statements, but it would decrease the indent by one. But it would also result in two separate `return loadScriptPromise` statements in the function, one at the beginning and one at the end:

```javascript
export function loadScript(locale) {
  if (loadScriptPromise) {
    return loadScriptPromise
  }

  // assignment of loadScriptPromise elided

  return loadScriptPromise
}
```

It's all a matter of style, so I'll go with whichever way you prefer. But I'll wait to hear back for your final decision before changing this one.

Reply:
I prefer multiple returns instead of else statements. But in that cae, as pointed by you, the ternary looks way better.